### ISSUE TITLE
Ros install takagi/lake does not work

### ISSUE THREAD

**Issue by svetlyak40wt** [takagi/lake#90]
Here is what I've got today, trying to install it on a fresh virtual machine in the cloud:

```
art@cl122dfojq09vbc03vn8-yhub:~$ ros install takagi/lake
Installing from github takagi/lake
[1/3] System 'lake' found. Loading the system..
While evaluating the form starting at line 11, column 0
  of #P"/home/art/.roswell/local-projects/takagi/lake/lake.asd":Aborted during step [1/3].
Unhandled MISSING-DEPENDENCY in thread #<SB-THREAD:THREAD "main thread" RUNNING {1004A90173}>: Component "deploy" not found, required by NIL

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {1004A90173}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK Component "deploy" not found, required by NIL #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK SB-EXT:*INVOKE-DEBUGGER-HOOK* Component "deploy" not found, required by NIL)
2: (INVOKE-DEBUGGER Component "deploy" not found, required by NIL)
3: (ERROR MISSING-DEPENDENCY :REQUIRED-BY NIL :REQUIRES "deploy")
4: (ASDF/FIND-COMPONENT:RESOLVE-DEPENDENCY-NAME NIL "deploy" NIL)
5: ((LAMBDA NIL :IN ASDF/PARSE-DEFSYSTEM:REGISTER-SYSTEM-DEFINITION))
6: (SB-INT:SIMPLE-EVAL-IN-LEXENV (DEFSYSTEM "lake" :VERSION "0.1.3" :AUTHOR "Rudolph Miller and Masayuki Takagi" :LICENSE "MIT" :HOMEPAGE "https://github.com/takagi/lake" :CLASS :PACKAGE-INFERRED-SYSTEM ...) #<NULL-LEXENV>)
7: (SB-EXT:EVAL-TLF (DEFSYSTEM "lake" :VERSION "0.1.3" :AUTHOR "Rudolph Miller and Masayuki Takagi" :LICENSE "MIT" :HOMEPAGE "https://github.com/takagi/lake" :CLASS :PACKAGE-INFERRED-SYSTEM ...) 3 NIL)
8: ((LABELS SB-FASL::EVAL-FORM :IN SB-INT:LOAD-AS-SOURCE) (DEFSYSTEM "lake" :VERSION "0.1.3" :AUTHOR "Rudolph Miller and Masayuki Takagi" :LICENSE "MIT" :HOMEPAGE "https://github.com/takagi/lake" :CLASS :PACKAGE-INFERRED-SYSTEM ...) 3)
9: ((LAMBDA (SB-KERNEL:FORM &KEY :CURRENT-INDEX &ALLOW-OTHER-KEYS) :IN SB-INT:LOAD-AS-SOURCE) (DEFSYSTEM "lake" :VERSION "0.1.3" :AUTHOR "Rudolph Miller and Masayuki Takagi" :LICENSE "MIT" :HOMEPAGE "https://github.com/takagi/lake" :CLASS :PACKAGE-INFERRED-SYSTEM ...) :CURRENT-INDEX 3)
10: (SB-C::%DO-FORMS-FROM-INFO #<FUNCTION (LAMBDA (SB-KERNEL:FORM &KEY :CURRENT-INDEX &ALLOW-OTHER-KEYS) :IN SB-INT:LOAD-AS-SOURCE) {1003D9B78B}> #<SB-C::SOURCE-INFO {1003D9B753}> SB-C::INPUT-ERROR-IN-LOAD)
11: (SB-INT:LOAD-AS-SOURCE #<SB-INT:FORM-TRACKING-STREAM for "file /home/art/.roswell/local-projects/takagi/lake/lake.asd" {1003D99CD3}> :VERBOSE NIL :PRINT NIL :CONTEXT "loading")
12: ((LABELS SB-FASL::LOAD-STREAM-1 :IN LOAD) #<SB-INT:FORM-TRACKING-STREAM for "file /home/art/.roswell/local-projects/takagi/lake/lake.asd" {1003D99CD3}> NIL)
13: (SB-FASL::CALL-WITH-LOAD-BINDINGS #<FUNCTION (LABELS SB-FASL::LOAD-STREAM-1 :IN LOAD) {7FCCD045D9FB}> #<SB-INT:FORM-TRACKING-STREAM for "file /home/art/.roswell/local-projects/takagi/lake/lake.asd" {1003D99CD3}> NIL #<SB-INT:FORM-TRACKING-STREAM for "file /home/art/.roswell/local-projects/takagi/lake/lake.asd" {1003D99CD3}>)
14: (LOAD #P"/home/art/.roswell/local-projects/takagi/lake/lake.asd" :VERBOSE NIL :PRINT NIL :IF-DOES-NOT-EXIST T :EXTERNAL-FORMAT :UTF-8)
15: (UIOP/UTILITY:CALL-WITH-MUFFLED-CONDITIONS #<FUNCTION (LAMBDA NIL :IN UIOP/LISP-BUILD:LOAD*) {1003D98C9B}> ("Overwriting already existing readtable ~S." #(#:FINALIZERS-OFF-WARNING :ASDF-FINALIZERS)))
16: ((FLET "THUNK" :IN PERFORM))
17: (SB-IMPL::%WITH-STANDARD-IO-SYNTAX #<FUNCTION (FLET "THUNK" :IN PERFORM) {7FCCD045DC5B}>)
18: ((:METHOD PERFORM (DEFINE-OP SYSTEM)) #<DEFINE-OP > #<ASDF/SYSTEM:UNDEFINED-SYSTEM "lake">) [fast-method]
19: ((SB-PCL::EMF PERFORM) #<unused argument> #<unused argument> #<DEFINE-OP > #<ASDF/SYSTEM:UNDEFINED-SYSTEM "lake">)
20: ((LAMBDA NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
21: ((:METHOD PERFORM-WITH-RESTARTS :AROUND (T T)) #<DEFINE-OP > #<ASDF/SYSTEM:UNDEFINED-SYSTEM "lake">) [fast-method]
22: ((:METHOD PERFORM-PLAN (T)) #<SEQUENTIAL-PLAN {1003D97A23}>) [fast-method]
23: ((FLET SB-C::WITH-IT :IN SB-C::%WITH-COMPILATION-UNIT))
24: ((:METHOD PERFORM-PLAN :AROUND (T)) #<SEQUENTIAL-PLAN {1003D97A23}>) [fast-method]
25: ((:METHOD OPERATE (OPERATION COMPONENT)) #<DEFINE-OP > #<ASDF/SYSTEM:UNDEFINED-SYSTEM "lake"> :PLAN-CLASS NIL :PLAN-OPTIONS NIL) [fast-method]
26: ((SB-PCL::EMF OPERATE) #<unused argument> #<unused argument> #<DEFINE-OP > #<ASDF/SYSTEM:UNDEFINED-SYSTEM "lake">)
27: ((LAMBDA NIL :IN OPERATE))
28: ((:METHOD OPERATE :AROUND (T T)) #<DEFINE-OP > #<ASDF/SYSTEM:UNDEFINED-SYSTEM "lake">) [fast-method]
29: ((LAMBDA NIL :IN FIND-SYSTEM))
30: (ASDF/SESSION:CONSULT-ASDF-CACHE (FIND-SYSTEM "lake") #<FUNCTION (LAMBDA NIL :IN FIND-SYSTEM) {1003D9507B}>)
31: ((:METHOD FIND-COMPONENT (STRING T)) "lake" (NIL) :REGISTERED NIL) [fast-method]
32: ((:METHOD OPERATE (OPERATION T)) #<DEFINE-OP > ("lake")) [fast-method]
33: ((SB-PCL::EMF OPERATE) #<unused argument> #<unused argument> #<DEFINE-OP > ("lake"))
34: ((LAMBDA NIL :IN OPERATE))
35: ((:METHOD OPERATE :AROUND (T T)) #<DEFINE-OP > ("lake")) [fast-method]
36: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<FUNCTION (LAMBDA NIL :IN OPERATE) {1003D94B9B}> :OVERRIDE T :KEY NIL :OVERRIDE-CACHE T :OVERRIDE-FORCING NIL)
37: ((LAMBDA NIL :IN OPERATE))
38: ((:METHOD OPERATE :AROUND (T T)) #<DEFINE-OP > #<ASDF/SYSTEM:UNDEFINED-SYSTEM "lake">) [fast-method]
39: ((LAMBDA NIL :IN LOAD-ASD))
40: ((LAMBDA NIL :IN FIND-SYSTEM))
41: (ASDF/SESSION:CONSULT-ASDF-CACHE (FIND-SYSTEM "lake") #<FUNCTION (LAMBDA NIL :IN FIND-SYSTEM) {1003D82DDB}>)
42: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<FUNCTION (LAMBDA NIL :IN FIND-SYSTEM) {1003D82DDB}> :OVERRIDE NIL :KEY (FIND-SYSTEM "lake") :OVERRIDE-CACHE NIL :OVERRIDE-FORCING NIL)
43: (QUICKLISP-CLIENT:WHERE-IS-SYSTEM "lake")
44: (ROSWELL.INSTALL:INSTALL-SYSTEM-SCRIPT "lake")
45: (ROSWELL.INSTALL:INSTALL-SYSTEM-IF-PROBED "lake")
46: (ROSWELL.INSTALL:INSTALL NIL)
47: (SB-INT:SIMPLE-EVAL-IN-LEXENV (APPLY (QUOTE ROS.SCRIPT.INSTALL.3671936092::MAIN) ROSWELL:*ARGV*) #<NULL-LEXENV>)
48: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:QUIT (APPLY (QUOTE ROS.SCRIPT.INSTALL.3671936092::MAIN) ROSWELL:*ARGV*)) #<NULL-LEXENV>)
49: (SB-EXT:EVAL-TLF (ROSWELL:QUIT (APPLY (QUOTE ROS.SCRIPT.INSTALL.3671936092::MAIN) ROSWELL:*ARGV*)) NIL NIL)
50: ((LABELS SB-FASL::EVAL-FORM :IN SB-INT:LOAD-AS-SOURCE) (ROSWELL:QUIT (APPLY (QUOTE ROS.SCRIPT.INSTALL.3671936092::MAIN) ROSWELL:*ARGV*)) NIL)
51: (SB-INT:LOAD-AS-SOURCE #<CONCATENATED-STREAM :STREAMS NIL {100342B7D3}> :VERBOSE NIL :PRINT NIL :CONTEXT "loading")
52: ((LABELS SB-FASL::LOAD-STREAM-1 :IN LOAD) #<CONCATENATED-STREAM :STREAMS NIL {100342B7D3}> NIL)
53: (SB-FASL::CALL-WITH-LOAD-BINDINGS #<FUNCTION (LABELS SB-FASL::LOAD-STREAM-1 :IN LOAD) {7FCCD045F6AB}> #<CONCATENATED-STREAM :STREAMS NIL {100342B7D3}> NIL #<CONCATENATED-STREAM :STREAMS NIL {100342B7D3}>)
54: (LOAD #<CONCATENATED-STREAM :STREAMS NIL {100342B7D3}> :VERBOSE NIL :PRINT NIL :IF-DOES-NOT-EXIST T :EXTERNAL-FORMAT :DEFAULT)
55: ((FLET ROSWELL::BODY :IN ROSWELL:SCRIPT) #<SB-SYS:FD-STREAM for "file /usr/local/etc/roswell/install.ros" {1003423C03}>)
56: (ROSWELL:SCRIPT "/usr/local/etc/roswell/install.ros" "takagi/lake")
57: (ROSWELL:RUN ((:EVAL "(ros:quicklisp)") (:SCRIPT "/usr/local/etc/roswell/install.ros" "takagi/lake") (:QUIT NIL)))
58: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:RUN (QUOTE ((:EVAL "(ros:quicklisp)") (:SCRIPT "/usr/local/etc/roswell/install.ros" "takagi/lake") (:QUIT NIL)))) #<NULL-LEXENV>)
59: (EVAL (ROSWELL:RUN (QUOTE ((:EVAL "(ros:quicklisp)") (:SCRIPT "/usr/local/etc/roswell/install.ros" "takagi/lake") (:QUIT NIL)))))
60: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:EVAL . "(progn #-ros.init(cl:load \"/usr/local/etc/roswell/init.lisp\"))") (:EVAL . "(ros:run '((:eval\"(ros:quicklisp)\")(:script \"/usr/local/etc/roswell/install.ros\"\"takagi/lake\")(:quit ())))")))
61: (SB-IMPL::TOPLEVEL-INIT)
62: ((FLET SB-UNIX::BODY :IN SB-IMPL::START-LISP))
63: ((FLET "WITHOUT-INTERRUPTS-BODY-3" :IN SB-IMPL::START-LISP))
64: (SB-IMPL::START-LISP)
```

**Comment by svetlyak40wt** [comment]
## A workaround

Do install `deploy` as a separate step:

```
ros install deploy
ros install takagi/lake
```

**Comment by takagi** [comment]
Thanks, I have replicated the situation. Do you have any idea to fix this? (sorry, I don't have time to work on this)

**Comment by svetlyak40wt** [comment]
Seems this problem is in the Roswell itself. It is able to install Lake with it's dependencies from Quicklisp, but not from the Github.

Easiest way to check is to run `ros install` under the `qlot` environment:

## This will work

```
rm -fr .qlot qlfile qlfile.lock
touch qlfile
qlot install

qlot exec ros install lake
```

Seems, this work because `ros` uses `quicklisp` client to download `lake` and its dependencies. And QL client sees there is `deploy` dependency in the metadata. 

## But this does not

```
rm -fr .qlot qlfile qlfile.lock
touch qlfile
qlot install

qlot exec ros install takagi/lake
```

Probably because `ros` just clones git repository to the `local-projects` dir and calls `(asdf:load-system :lake)` which does not download any dependency.

**Comment by svetlyak40wt** [comment]
I've did this fix: https://github.com/takagi/lake/pull/91

**Comment by takagi** [comment]
Closed via #91.
